services-optimization/src/test/java/com/mapbox/api/optimization/v1/OptimizationResponseFactoryTest.java

```diff
@@ -0,0 +1,271 @@
+package com.mapbox.api.optimization.v1;
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.RouteOptions;
+import com.mapbox.api.optimization.v1.models.OptimizationResponse;
+import com.mapbox.api.optimization.v1.models.OptimizationWaypoint;
+import okhttp3.Headers;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import retrofit2.Response;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class OptimizationResponseFactoryTest {
+
+  @Mock
+  private MapboxOptimization optimization;
+  private OptimizationResponseFactory factory;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.openMocks(this);
+    factory = new OptimizationResponseFactory(optimization);
+  }
+
+  @Test
+  public void generate_successful() {
+    final String profile = "driving-traffic";
+    final String coordinates = "1.2,3.4,5.6,7.8";
+    final String annotations = "congestion,duration";
+    final String bearings = "123,456";
+    final String language = "it";
+    final String radiuses = "10,20";
+    final String user = "some_user";
+    final String overview = "full";
+    final boolean steps = true;
+    final String baseUrl = "some url";
+    final String geometries = "polyline,polyline6";
```

Comment:
```suggestion
    final String geometries = "polyline6";
```
nit: You can provide only one geometry.